rqt_py_common/src/rqt_py_common/message_field_type_helpers.py

```diff
@@ -0,0 +1,397 @@
+from python_qt_binding.QtCore import qWarning
+
+from rosidl_parser.definition import BASIC_TYPES, INTEGER_TYPES, \
+                                     FLOATING_POINT_TYPES, CHARACTER_TYPES, \
+                                     BOOLEAN_TYPE, OCTET_TYPE
+
+MsgType = TypeVar('MsgType')
+
+BASIC_STRING_TYPES = (
+    'string',
+    'wstring'
+)
+
+PRIMITIVE_TYPES = (
+    *BASIC_TYPES,
+    *BASIC_STRING_TYPES
+)
+
+STRING_TYPES = (
+    *BASIC_STRING_TYPES,
+    *CHARACTER_TYPES
+)
+
+SEQUENCE_PREFIX = 'sequence<'
+SEQUENCE_PREFIX_LEN = len(SEQUENCE_PREFIX)
+
+__LOGGER = logging.get_logger('message_field_type_helpers')
+
+class MessageFieldTypeInfo(object):
+    __slots__ = (
+        'base_type_str',
+        'is_array',
+        'is_static_array',
+        'static_array_size',
+        'is_bounded_array',
+        'bounded_array_size',
+        'is_unbounded_array',
+        'is_bounded_string',
+        'bounded_string_size',
+    )
+
+    """
+    Useful helper methods for working with a message's field types
+
+    Parse items such as: (from rosidl_generator_py definitions)
+      .msg            field_type                    python
+      -----------------------------------------------------------
+      bool            -> 'boolean'                  -> bool
+      byte            -> 'octet'                    -> bytes
+      float32         -> 'float'                    -> float
+      float64         -> 'double'                   -> float
+      char            -> 'uint8'                    -> int
+      int8            -> 'int8'                     -> int
+      int16           -> 'int16'                    -> int
+      int32           -> 'int32'                    -> int
+      int64           -> 'int64'                    -> int
+      uint8           -> 'uint8'                    -> int
+      uint16          -> 'uint16'                   -> int
+      uint32          -> 'uint32'                   -> int
+      uint64          -> 'uint64'                   -> int
+      string          -> 'string'                   -> str
+      my_msgs/Custom  -> 'my_msgs/Custom'           -> my_msgs.msg.Custom
+
+      # Arrays with normal values
+      int8[3]         -> 'int8[3]'                  -> List[int]
+      int8[]          -> 'sequence<int8>'           -> List[int]
+      int8[<=3]       -> 'sequence<int8, 3>'        -> List[int]
+
+      # String weirdness
+      string<=5       -> 'string<5>'                -> str
+      string[3]       -> 'string[3]'                -> List[str]
+      string[]        -> 'sequence<string>'         -> List[str]
+      string<=5[3]    -> 'string<5>[3]'             -> List[str]
+      string<=5[<=10] -> 'sequence<string<5>, 10>'  -> List[str]
+      string<=5[]     -> 'sequence<string<5>>'      -> List[str]
+      string[<=10]    -> 'sequence<string, 10>'     -> List[str]
+
+    :param field_type: the field type in the form generated by msg.get_fields_and_field_types
+    :type field_type: str
+    """
```

Comment:
I found this to be really really helpful when trying to understand the field_type parsing